src/Cli/dotnet/AutomaticEncodingRestorer.cs

```diff
@@ -0,0 +1,65 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+//
+
+using System;
+using System.IO;
+using System.Security;
+using System.Text;
+
+namespace Microsoft.DotNet.Cli
+{
+    /// <summary>
+    /// A program can change the encoding of the console which would affect other programs.
```

Comment:
Are there links to any documentation we can include in the code?

Reply:
Unfortunately no.
But here is some evidence that this is true, besides the customer who reported it as an issue:

![image](https://user-images.githubusercontent.com/23152278/223573419-cd1df883-209b-4b8e-b5d3-e9dad571b67a.png)